packages/expo-modules-autolinking/scripts/ios/react_import_patcher.rb

```diff
@@ -0,0 +1,72 @@
+# Copyright 2018-present 650 Industries. All rights reserved.
+
+module Expo
+  class ReactImportPatcher
+
+    public def initialize(installer, options)
+      @root = installer.sandbox.root
+      @module_dirs = get_module_dirs(installer)
+      @options = options
+    end
+
+    public def run!
+      args = [
+        'node',
+        '--eval',
+        'require(\'expo-modules-autolinking\')(process.argv.slice(1))',
+        'patch-react-imports',
+        '--pods-root',
+        Shellwords.escape(File.expand_path(@root)),
+      ]
+
+      if @options[:dry_run]
+        args.append('--dry-run')
+      end
+
+      @module_dirs.each do |dir|
+        args.append(Shellwords.escape(File.expand_path(dir)))
+      end
+      Pod::UI.message "Executing ReactImportsPatcher node command: #{args.join(' ')}"
+
+      time_begin = Process.clock_gettime(Process::CLOCK_MONOTONIC)
+      system(*args)
+      elapsed_time = Process.clock_gettime(Process::CLOCK_MONOTONIC) - time_begin
+      Pod::UI.info "ReactImportsPatcher takes #{elapsed_time.round(4)} seconds to transform files."
+    end
+
+    private def get_module_dirs(installer)
+      unless installer.pods_project
+        Pod::UI.message '`pods_project` not found. This happens possibly from `incremental_installation`'
+        return []
+      end
+
+      result = []
+      installer.pods_project.development_pods.children.each do |pod|
+        if pod.is_a?(Xcodeproj::Project::Object::PBXFileReference) && pod.path.end_with?('.xcodeproj')
+          # Support generate_multiple_pod_projects or use_frameworks!
+          project = Xcodeproj::Project.open(File.join(installer.sandbox.root, pod.path))
+          groups = project.groups.select { |group| !(['Dependencies', 'Frameworks', 'Products'].include? group.name) }
+          groups.each do |group|
+            unless group.path.start_with? '../'
+              raise 'CocoaPods does not put pod subprojects inside nested directory'
+            end
+            result.append(group.path[3..])
+          end
+        else
+          unless pod.path.start_with? '../'
+            raise 'CocoaPods does not put pod subgroups inside nested directory'
+          end
+          result.append(pod.path[3..])
+        end
+      end
+
+      return result.select { |dir|
+        # Exclude known dirs unnecessary to patch and reduce processing time
+        !dir.include?('/react-native/') &&
+        !dir.end_with?('/react-native') &&
+        !dir.include?('/expo-')
```

Comment:
This is probably fine, but might break if someone made a module named `expo-<their module>` with `#import "RCTBridge.h"`

Reply:
yes, it's a good time to know if there're any other third-party `expo-` prefixed modules and break the double-quoted thing. then i'll send a pr for them in this case.

Reply:
In my case, out actual top-level project was named `react-native`, which caused some issues when running this script. We've renamed out project, but thought I would drop this here in case anyone is facing issues here.

Reply:
thanks for digging the issue. react-native has a lot double-quoted imports. they're internal imports so it works fine.  transforming all the internal import is risky imo. that's the main reason i'd exclude the path here.